exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRoot.java

```diff
@@ -55,44 +59,89 @@ public Viewable getStats() {
+    final String address = currentEndpoint.getAddress();
+    props.put("Data Port Address", address + ":" + currentEndpoint.getDataPort());
+    props.put("User Port Address", address + ":" + currentEndpoint.getUserPort());
+    props.put("Control Port Address", address + ":" + currentEndpoint.getControlPort());
+    props.put("Maximum Direct Memory", DrillConfig.getMaxDirectMemory());
+
+    return new Stats(props, collectDrillbits(version));
+  }
+
+  private Collection<DrillbitInfo> collectDrillbits(String version) {
+    Set<DrillbitInfo> drillbits = Sets.newTreeSet();
+    for (CoordinationProtos.DrillbitEndpoint endpoint : work.getContext().getBits()) {
+      boolean versionMatch = version.equals(endpoint.getVersion());
+      DrillbitInfo drillbit = new DrillbitInfo(endpoint.getAddress(), endpoint.isInitialized(), endpoint.getVersion(), versionMatch);
+      drillbits.add(drillbit);
     }
-    stats.add(new Stat("Data Port Address", work.getContext().getEndpoint().getAddress() +
-      ":" + work.getContext().getEndpoint().getDataPort()));
-    stats.add(new Stat("User Port Address", work.getContext().getEndpoint().getAddress() +
-      ":" + work.getContext().getEndpoint().getUserPort()));
-    stats.add(new Stat("Control Port Address", work.getContext().getEndpoint().getAddress() +
-      ":" + work.getContext().getEndpoint().getControlPort()));
-    stats.add(new Stat("Maximum Direct Memory", DrillConfig.getMaxDirectMemory()));
-
-    return stats;
+    return drillbits;
   }
 
   @XmlRootElement
-  public class Stat {
-    private String name;
-    private Object value;
+  public static class Stats {
+    private final Map<String, Object> props;
+    private final Collection<DrillbitInfo> drillbits;
 
     @JsonCreator
-    public Stat(String name, Object value) {
-      this.name = name;
-      this.value = value;
+    public Stats(Map<String, Object> props, Collection<DrillbitInfo> drillbits) {
+      this.props = props;
+      this.drillbits = drillbits;
+    }
+
+    public Map<String, Object> getProps() {
+      return props;
     }
 
-    public String getName() {
-      return name;
+    public Collection<DrillbitInfo> getDrillbits() {
+      return drillbits;
     }
+  }
+
+  public static class DrillbitInfo implements Comparable<DrillbitInfo> {
+    private final String address;
+    private final boolean initialized;
+    private final String version;
+    private final boolean versionMatch;
 
-    public Object getValue() {
-      return value;
+    @JsonCreator
+    public DrillbitInfo(String address, boolean initialized, String version, boolean versionMatch) {
+      this.address = address;
+      this.initialized = initialized;
+      this.version = version;
+      this.versionMatch = versionMatch;
     }
 
+    public String getAddress() {
+      return address;
+    }
+
+    public String isInitialized() {
+      return initialized ? "initialized" : "not initialized";
```

Comment:
boolean-to-string conversions are best handled in the web UI template so that they can possibly be localized later. And, all the user-visible text is in one place.